src/Umbraco.Web/PropertyEditors/MediaPicker2PropertyEditor.cs

```diff
@@ -35,12 +35,19 @@ internal class MediaPickerPreValueEditor : PreValueEditor
         {
             public MediaPickerPreValueEditor()
             {
-                //create the fields
                 Fields.Add(new PreValueField()
                 {
-                    Key = "multiPicker",
```

Comment:
@skartknet As taken from review notes here http://issues.umbraco.org/issue/U4-4282

> What I don't understand is the upgrade path: the property editor replaces the  multiPicker  preValue with minNumber  and maxNumber  preValues, which will both (on upgrade) be un-initialized, and then how is the controller going to figure out whether it deals with a multi-picker or a simple picker?

This is true, I think what needs to happen is either:

* The `multiPicker` pre-value needs to remain and the description of it needs to be updated to indicate that the min/max values are ignored if this is not checked. The actual property editor would also need to know how to take into account all of this data since on upgrade the 'multiPicker' pre-value will still be there and if detected, the property editor will need to translate this into the min/max values.

OR (a nicer option)

* A new pre-value editor should be created which encapsulates all of these fields and the min/max number fields are disabled unless the 'pick multiple items' is checked. This new pre-value editor can go a step further and take into account this 'multiPicker' value and if it's defined and it's set to false, then the max value can be set to 1 which would be basically equivalent, else it could be set to -1 which would mean you can pick any number. The actual property editor would also need to know how to take into account all of this data since on upgrade the 'multiPicker' pre-value will still be there and if detected, the property editor will need to translate this into the min/max values.

thoughts?

Reply:
You're correct, I didn't think about upgrading from current values :(. I really like the second option of creating a new pre-value editor, it gives a better user experience. I guess it could also help to improve the the Multinode Treepicker as well?